transaction/builder/src/memo_builder/rth_memo_builder.rs

```diff
@@ -76,12 +75,44 @@ pub struct RTHMemoBuilder {
     fee: Amount,
 }
 
+#[derive(Clone, Debug)]
+pub enum CustomMemoType {
+    PaymentRequestId(u64),
+    PaymentIntentId(u64),
+    FlexibleMemos(FlexibleMemoPayloads),
+}
+
+/// This contains both the output memo payload and the change memo payload which
+/// will be used when generating custom output memos and change memos
+#[derive(Clone, Debug)]
+pub struct FlexibleMemoPayloads {
+    /// This is used when generating output memos.
+    /// It must contain a memo type with the first byte being 0x01
+    pub output_memo_payload: FlexibleMemoPayload,
+    /// This is used when generating change memos.
+    /// It must contain a memo type with the first byte being 0x02
+    pub change_memo_payload: FlexibleMemoPayload,
+}
+
+/// This is the payload data used for creating custom MemoPayloads with types
+/// which are defined in MCIPs.
+#[derive(Clone, Debug)]
+pub struct FlexibleMemoPayload {
+    /// Corresponds to the returned memo type. This should be
+    /// listed in an MCIP.
+    pub memo_type_bytes: [u8; 2],
```

Comment:
this is wrong, it needs to be just `second_memo_type_byte: u8`, and the only allowed first byte is 0x01, for the authenticated sender category, as described in #3200